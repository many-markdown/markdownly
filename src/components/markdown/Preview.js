import React from 'react';
import PropTypes from 'prop-types';
import marked from 'marked';
import styles from 'github-markdown-css/github-markdown.css';

const Preview = ({ markdown }) => {
  const __html = marked(markdown);
  return <div className={styles['markdown-body']} dangerouslySetInnerHTML={{ __html }}></div>;
};

Preview.propTypes = {
  markdown: PropTypes.string.isRequired
};

export default Preview;
