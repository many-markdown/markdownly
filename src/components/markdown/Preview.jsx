import React from 'react';
import PropTypes from 'prop-types';
import marked from 'marked';
import styles from 'github-markdown-css/github-markdown.css';
import { useSelector } from '../../hooks/appContext';
import { getMarkdown } from '../../selectors/appSelectors';

const Preview = () => {
  const markdown = useSelector(getMarkdown);
  const __html = marked(markdown);
  return <div className={styles['markdown-body']} dangerouslySetInnerHTML={{ __html }}></div>;
};

// Preview.propTypes = {
//   markdown: PropTypes.string.isRequired
// };

export default Preview;
