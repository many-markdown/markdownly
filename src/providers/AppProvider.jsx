import React, { useReducer, useEffect } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { setInitialState } from '../actions/markdownActions';

// eslint-disable-next-line react/prop-types
export default function AppProvider({ children }) {

  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    const initialState = localStorage.getItem('FILE_STORAGE'); 
    dispatch(setInitialState(JSON.parse(initialState)));
  }, []);

  return (
    <AppContext.Provider value={{ state, dispatch }} >
      {children}
    </AppContext.Provider>
  );
}
