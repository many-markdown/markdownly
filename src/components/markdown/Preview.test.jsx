import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview.jsx';

import AppProvider from '../../providers/AppProvider';




describe('Preview component', () => {
  
  it('renders Preview', () => {
    const wrapper = shallow(<AppProvider><Preview /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
