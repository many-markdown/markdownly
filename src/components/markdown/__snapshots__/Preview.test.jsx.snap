// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component renders Preview 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "markdown": "# hello there",
      },
    }
  }
>
  <Preview />
</ContextProvider>
`;
